File: 1. Creational Design Patterns/Singleton/C#/SingletonTemplate.cs
// Singleton Pattern – C# Template
using System;

sealed class Singleton
{
    private static readonly Lazy<Singleton> _instance = new(() => new Singleton());
    public static Singleton Instance => _instance.Value;
    private Singleton() { }
}

class Entry { static void Main(){ var s = Singleton.Instance; } }

File: 1. Creational Design Patterns/Singleton/Typescript/SingletonTemplate.ts
// Singleton Pattern – TypeScript Template
// Simple lazy-initialized singleton without external typings.

declare const console: any;

class Singleton {
  private static _instance: Singleton | null = null;
  private constructor() {}
  static getInstance(): Singleton {
    return this._instance ?? (this._instance = new Singleton());
  }
}

// Usage:
// const s = Singleton.getInstance();
